### PR TITLE
Update to OTLP proto v0.18

### PR DESCRIPTION
**Pull request by MikeGoldsmith**
## Which problem is this PR solving?

Updates the OTLP proto spec to latest (currently v0.18). This includes moving past a version where previously supported deprecated trace status' were allowed. This PR updates the logic to only use the newer status codes.

- Closes https://github.com/honeycombio/telemetry-team/issues/240

## Short description of the changes
- Update otlp-proto package to 0.18
- Remove deprecated status code logic when processing spans
- Update unit tests for deprecated span status codes

### CODE REVIEW

**Comment by robbkidd** [comment]
Blocking merge and release until log support (#82) merges and is released.

**Comment by pkanal** [comment]
> Blocking merge and release until log support (#82) merges and is released.
> 
> Merge is also blocked until [transform of OTLP Metrics moves into husky](https://github.com/honeycombio/telemetry-team/issues/290). OTLP proto update and Metrics transform will travel together.

After working through it, we'll actually have to merge this PR before metrics can be moved into Husky. The order of operations will be:
1. Upgrade OTLP proto to v0.18 in Husky and release Husky (this PR)
2. Upgrade OTLP proto to v0.18 in Shepherd and upgrade to latest Husky and drop support for deprecated fields (https://github.com/honeycombio/hound/pull/10705)
3. Add metrics logic to Husky in a separate PR and release a new version of Husky
4. Update Shepherd to use Husky for metrics logic

**Comment by MikeGoldsmith** [comment]
Unlocking now Logs has been used in Shepherd for a little while and we're confident it won't just break.

File: otlp/traces.go
Comment (kentquirk) [inline otlp/traces.go:258]: This entire function reduces to:

```go
func evaluateSpanStatus(status *trace.Status) (int, bool) {
	if status == nil {
		return int(trace.Status_STATUS_CODE_UNSET), false
	}

	return int(status.Code), false
}
```

And since it never returns true, should probably also have the extra bool parameter removed.

File: otlp/traces_test.go
Comment (kentquirk) [inline otlp/traces_test.go:1200]: Message should be something like "Error!"

File: otlp/traces_test.go
Comment (kentquirk) [inline otlp/traces_test.go:1191]: Message should be something like "Unset!"

File: otlp/traces_test.go
Comment (kentquirk) [inline otlp/traces_test.go:1209]: Not "Old" anymore.

**Review by kentquirk — COMMENTED** [review]
Since the old cruft is gone from upstream, we should remove it from here as well.

File: otlp/traces.go
Comment (MikeGoldsmith) [reply otlp/traces.go:258]: We still want to return the bool if the status code is ERROR, but I agree this func is a little more complicated than necessary where it makes local constants of the ok, error and unset statuses. I'll refactor to make simpler.

**Review by kentquirk — APPROVED** [review]
tenor-83316687

**Pull request merged**
